h5io/tests/test_io.py

```diff
@@ -38,6 +38,27 @@ def test_hdf5():
                   overwrite=True)
     assert_raises(TypeError, write_hdf5, test_file, object, overwrite=True)
 
+    write_hdf5(test_file, 1, title='first')
+    write_hdf5(test_file, 2, title='second')
```

Comment:
...please restore it to its old behavior, and add a test that it does behave that way.